iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlPreprocessorTest.java

```diff
@@ -68,4 +68,29 @@ void shouldRemoveInlineCommentsAfterDocumentBreak() {
       """;
     assertEquals(expected, new SpringConfigYamlPreprocessor().preprocess(code));
   }
+
+  @Test
+  void shouldTransformMavenSubstitutions() {
+    // language=yaml
+    var code = """
+      foo: @maven.property@
+      foo2: @maven.property@-suffix
+      foo3: '@maven.property@'
+      bar: prefix-@maven.property@-suffix
+      baz: |
+        @maven.property@
+      """;
+    // language=yaml
+    var expected = """
+      foo: 'maven.property'
+      foo2: 'maven.property'-suffix
+      foo3: '@maven.property@'
+      bar: prefix-@maven.property@-suffix
```

Comment:
What about a line with two variables?
```
      bar: prefix-@maven.property1@-@maven.property2@
```

Reply:
Good catch, I'll add a test case. The idea is to only substitute the first variable in a line, because when `@` is in the middle of a value, snakeyaml doesn't have a problem with it.